### PR TITLE
Check claimRef UID when processing a recycled PV

### PR DESCRIPTION
**Pull request by ghost**
After a volume is recycled and it is in the pending state if a new PVC is created with the same name that the old PV was bound to, the newly recycled PV never gets to the Available state. This patch corrects that
@jsafrane WDYT ?

### CODE REVIEW

**Comment by jsafrane** [comment]
Good catch, LGTM.

File: pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go [hunk 1]
```diff
@@ -238,7 +238,7 @@ func syncVolume(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCl
 
 		if volume.Spec.ClaimRef != nil {
 			claim, err := binderClient.GetPersistentVolumeClaim(volume.Spec.ClaimRef.Namespace, volume.Spec.ClaimRef.Name)
-			if errors.IsNotFound(err) {
+			if errors.IsNotFound(err) || (claim != nil && claim.UID != volume.Spec.ClaimRef.UID) {
```
Comment (saad-ali) [inline pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go]: nit: log this for debugging.

**Comment by saad-ali** [comment]
LGTM mod nit

File: pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go (hunk 1 above)
Comment (saad-ali) [inline pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go]: Can add in a follow up PR (I'm not going to remove LGTM).

**Comment by pmorie** [comment]
@swagiaal tests failed

**Comment by ncdc** [comment]
build timed out

@k8s-bot unit test this: #IGNORE

**Pull request merged**